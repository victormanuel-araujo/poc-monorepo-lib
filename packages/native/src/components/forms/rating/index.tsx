export * from './rating.comp'
export * from './rating.types'
