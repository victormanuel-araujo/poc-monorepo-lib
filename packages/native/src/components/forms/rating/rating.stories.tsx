import React from 'react'

import { Rating, RatingProps } from '@components'
import { Meta, Story } from '@storybook/react'

export default {
  title: 'Forms/Rating',
  component: Rating,
} as Meta

const Template: Story<RatingProps> = (args) => <Rating {...args} />

export const Default = Template.bind({})
Default.args = {
  disabled: false,
  maxRatingValue: 5,
  value: 0,
  showTotalOfEvaluators: false,
  totalOfEvaluators: 0,
}
