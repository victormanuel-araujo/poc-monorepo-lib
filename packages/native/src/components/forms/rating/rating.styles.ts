import { View, TouchableOpacity } from 'react-native'
import styled from 'styled-components/native'

export const RatingContainer = styled(View)`
  flex-direction: row;
  align-items: center;
`

export const RatingButton = styled(TouchableOpacity)``
