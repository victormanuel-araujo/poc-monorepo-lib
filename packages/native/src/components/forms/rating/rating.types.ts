import { MaterialIconType } from '@components'
import { ThemeColorsName } from '@theme'

export interface RatingProps {
  /**
   * The highest possible rating value,
   * which reflects the number of rendered assessment buttons
   *
   * @default 5
   */
  maxRatingValue?: number

  /**
   * Define an initial value to the input
   *
   * Default is `0`
   */
  value?: number

  /**
   * Enable and Disable the rating editing.
   * If it is `true`, the value of the rating component cannot be clicked
   *
   * Default is `false`
   */
  disabled?: boolean

  /**
   * Callback that is called when the rating changes.
   */
  onChange?: (newValue: number) => void

  /**
   * Controls whether the number of reviewers should appear or not.
   * If 'true' the number entered through the props 'numberOfEvaluators' will be displayed in parentheses to the right of the last icon.
   */
  showTotalOfEvaluators?: boolean

  /**
   * Number that will be displayed in parentheses to the right of the last icon.
   */
  totalOfEvaluators?: number

  /**
   * Icons size in px
   */
  iconSize?: number
  /**
   * Material icon name to render as empty value icon
   */
  emptyIconName?: MaterialIconType
  /**
   *  Material icon name to render as half value icon
   */
  halfIconName?: MaterialIconType
  /**
   * Material icon name to render as rated value icon
   */
  fullIconName?: MaterialIconType
  /**
   * Color of the icons
   */
  iconColor?: ThemeColorsName
  /**
   * Background color of the icons
   */
  backgroundColor?: ThemeColorsName
}
