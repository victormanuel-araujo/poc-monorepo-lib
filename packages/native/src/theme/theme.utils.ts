// utils functions to handle theme
import { get, defaultsDeep } from 'lodash'
import { Platform } from 'react-native'
import { css } from 'styled-components/native'

import { theme as _theme } from './theme'
import {
  Theme,
  ThemeColorsName,
  ThemeColorsShade,
  ThemeFontWeights,
  ThemeFontSizes,
  ThemePaddings,
  ThemeProps,
  ThemeFonts,
} from './theme.types'

/**
 * This function create an theme object.
 * Optionally, you can pass an custom theme object with custom values and it will
 * always be merged with the default theme (smash)
 *
 * @param theme - custom theme
 *
 * @example
 * ```ts
 * const theme_a = { a: 1, b: 2, c: 3 };
 * const theme_b = { a: 2 };
 *
 * const theme_c = createTheme(theme_a, theme_b);
 *
 * // the output (theme_c) is equal to:
 * // theme_c = { a: 2, b: 2, c: 3 };
 * ```
 */
export const createTheme = (theme?: DeepPartial<Theme>): Theme => {
  return defaultsDeep(theme, _theme)
}

/**
 * easy way to get an theme color with a shade and optionally an alpha channel
 *
 * @param colorName - the name of the color
 * @param colorShade - the shade of the color
 * @param alpha - an alpha value in hexadecimal pattern, that is: two character from '00' to 'FF'
 *
 * @example
 * ```ts
 * inside a styled-component
 * ${getThemeColor(...props)}
 *
 * outside a styled-component
 * getThemeColor(...props)({ theme })
 * ```
 */
export const getThemeColor =
  (colorName: ThemeColorsName, colorShade: keyof ThemeColorsShade = 'base', alpha?: string) =>
  ({ theme }): string => {
    const color = get(theme, ['colors', colorName], { base: '#000000' })
    const shade: string = get(color, [colorShade], color.base)

    return alpha && alpha.length === 2 ? `${shade}${alpha}` : shade
  }

/**
 * Get the style of a font weight.
 * Verify the platform where the component is used, and return te correct style
 * to apply the font weight
 *
 * @param weight - Font weight, ie light, regular, bold, etc.
 * @param family - Font family - only used on Android
 *
 * @example
 * ```ts
 * const Bold = styled.Text`
 *   ${getFontWeight('bold')}
 * `
 * ```
 */
export const getFontWeight =
  (weight: keyof ThemeFontWeights, family: keyof ThemeFonts = 'default') =>
  ({ theme }) => {
    /*
     * In android, is necessary to change the font family too, cause the
     * font-weight property not work correctly
     */
    if (Platform.OS === 'android') {
      /* get the font family based on "family" prop  */
      const fontFamily: string = get(theme, ['fonts', family])

      /* return the style, with font-family and font-weight defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
        font-weight: ${get(theme, ['fontWeights', weight])};
      `
    }

    /*
     * In another platforms, as IOS or Web, is not necessary to set a new font family
     * so,
     */
    return css`
      font-weight: ${theme.fontWeights[weight]};
    `
  }

/**
 * Get the correct style for a font size.
 * return a style, with the font size and the relative line-height.
 *
 * @param size - Font size name, ie sm, md, lg, etc.
 * @param rawValue - If it's true, returns the raw integer value
 *                   instead of a style.
 *
 * @example
 * ```ts
 * const Title = styled.Text`
 *   ${getFontSize('lg')};
 * `
 * ```
 *
 * @example - `theme` comes from styled-component's ThemeContext.
 * ```tsx
 *   return <Icon size={getFontSize('lg', true)({ theme })} />
 * ```
 */
export const getFontSize =
  (size: keyof ThemeFontSizes, rawValue = false) =>
  ({ theme }: ThemeProps<Theme>) => {
    if (rawValue) {
      return get(theme, ['fontSizes', size])
    }

    return css`
      font-size: ${get(theme, ['fontSizes', size])}px;
      line-height: ${get(theme, ['lineHeights', size])}px;
    `
  }

/**
 * Used to get a style of a font family and verify the platform where the
 * component is used, and return te correct style
 * to apply the font family
 *
 * @param family - the name of the family
 * @param weight - optionally an weight of the family. use it only in android
 *
 * *
 * @example
 * ```ts
 * const TitleFamily = styled.Text`
 *   ${getFontFamily('title')}
 * `
 * ```
 */
export const getFontFamily =
  (family: keyof ThemeFonts = 'default', weight: keyof ThemeFontWeights = 'medium') =>
  ({ theme }: ThemeProps<Theme>) => {
    /* get the font family based on "family" prop  */
    const fontFamily: string = get(theme, ['fonts', family])

    if (Platform.OS === 'android') {
      /* return the style, with font-family defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
      `
    }

    return css`
      font-family: ${fontFamily};
    `
  }

/**
 * Get a named padding from current theme.
 *
 * @param size - Padding size name, ie sm, md, lg, etc.
 * @param multiply - used if is necessary to multiply the padding value
 *
 * @example
 * ```ts
 * const Container = styled.View`
 *   padding: ${getPadding('md')} ${getPadding('lg')};
 * `
 * ```
 */
export const getPadding =
  (size: keyof ThemePaddings, multiply = 1) =>
  ({ theme }: ThemeProps<Theme>): string => {
    return theme.paddings[size] * multiply + 'px'
  }
