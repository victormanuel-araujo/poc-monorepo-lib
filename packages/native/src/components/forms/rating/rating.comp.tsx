import React from 'react'

import { Icon, Text, MaterialIconType } from '@components'

import { RatingButton, RatingContainer } from './rating.styles'
import { RatingProps } from './rating.types'

/**
 * Rating is a simple component commonly used to rate something,
 * usually based on a number of stars.
 */
export const Rating: React.FC<RatingProps> = ({
  maxRatingValue,
  value,
  disabled,
  onChange,
  showTotalOfEvaluators,
  totalOfEvaluators,
  ...props
}) => {
  const [internalValue, setInternalValue] = React.useState<number>(value || 0)

  const { fullColorUntil, lastIconIsHalf } = React.useMemo(() => {
    const fullColorUntil = Math.floor(internalValue) || 0
    const lastIconIsHalf = internalValue - fullColorUntil >= 0.5
    return {
      fullColorUntil,
      lastIconIsHalf,
    }
  }, [internalValue])

  function renderIcon(iconIndex) {
    const lastFullColorIconIndex = fullColorUntil - 1
    const lastIconIndex = lastIconIsHalf ? lastFullColorIconIndex + 1 : lastFullColorIconIndex

    const { iconSize, emptyIconName, halfIconName, fullIconName, backgroundColor, iconColor } = props

    const renderIconByName = (name: MaterialIconType) => (
      <Icon.Material
        testID={`rating-icon-${iconIndex}`}
        icon={name}
        size={iconSize}
        background={backgroundColor}
        color={iconColor}
      />
    )

    let iconName = emptyIconName

    if (iconIndex <= lastFullColorIconIndex) {
      iconName = fullIconName
    } else if (iconIndex === lastIconIndex && lastIconIsHalf) {
      iconName = halfIconName
    }

    return renderIconByName(iconName)
  }

  function handleChange(newValue: number) {
    setInternalValue(newValue)

    if (onChange) {
      onChange(newValue)
    }
  }

  React.useEffect(() => {
    setInternalValue(value)
  }, [value])

  if (maxRatingValue <= 0) {
    return null
  }

  return (
    <RatingContainer testID="rating-container">
      {[...Array(maxRatingValue).keys()].map((_, index) => {
        const positon = index + 1
        return (
          <RatingButton
            testID={`rating-button-${index}`}
            disabled={disabled}
            key={`ratting-button-${index}`}
            onPress={() => handleChange(positon)}
          >
            {renderIcon(index)}
          </RatingButton>
        )
      })}
      {showTotalOfEvaluators && <Text testID="ratting-number-of-evaluators">({`${totalOfEvaluators || 0}`})</Text>}
    </RatingContainer>
  )
}

Rating.defaultProps = {
  maxRatingValue: 5,
  value: 0,
  disabled: false,
  onChange: null,
  iconSize: 20,
  emptyIconName: 'star-border',
  halfIconName: 'star-half',
  fullIconName: 'star',
  iconColor: 'yellow',
  backgroundColor: 'transparent',
  showTotalOfEvaluators: false,
  totalOfEvaluators: 0,
}
