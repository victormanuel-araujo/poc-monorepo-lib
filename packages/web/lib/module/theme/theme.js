import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from "./theme.values";
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */

export const theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings
};
//# sourceMappingURL=theme.js.map