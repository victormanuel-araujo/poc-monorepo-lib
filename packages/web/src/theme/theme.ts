import { Theme } from './theme.types'
import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */
export const theme: Theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings,
}
