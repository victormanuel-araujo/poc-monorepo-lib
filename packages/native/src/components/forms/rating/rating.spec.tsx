import React from 'react'

import { fireEvent } from '@testing-library/react-native'

import { renderWithTheme } from '../../../theme/theme.fixture'

import { Rating } from './rating.comp'

describe('Rating component', () => {
  it('should render correctly', () => {
    const component = renderWithTheme(<Rating />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render correctly with decimal numbers', () => {
    const component = renderWithTheme(<Rating value={2.5} />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should disable buttons', () => {
    const component = renderWithTheme(<Rating disabled maxRatingValue={2} />)

    expect(component.getByTestId('rating-button-0')).toBeDisabled()
    expect(component.getByTestId('rating-button-1')).toBeDisabled()
  })

  it('should call the onChange after user press', () => {
    const fn = jest.fn()
    const component = renderWithTheme(<Rating onChange={fn} />)
    const rattingButtonPositionOne = component.getByTestId('rating-button-1')

    fireEvent.press(rattingButtonPositionOne)

    expect(fn).toHaveBeenCalledWith(2)
  })

  it('should render number of evaluators', () => {
    const component = renderWithTheme(<Rating showTotalOfEvaluators totalOfEvaluators={100} />)
    const numberOfEvaluatorComponent = component.getByTestId('ratting-number-of-evaluators')
    expect(numberOfEvaluatorComponent.props.children.join('')).toBe('(100)')
  })
})
