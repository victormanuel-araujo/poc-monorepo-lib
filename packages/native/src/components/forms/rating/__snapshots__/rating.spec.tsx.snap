// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="rating-container"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-0"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-0"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-1"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-1"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-2"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-2"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-3"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-3"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-4"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-4"
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Rating component should render correctly with decimal numbers 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="rating-container"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-0"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-0"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-1"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-1"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-2"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-2"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-3"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-3"
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="rating-button-4"
  >
    <View
      background="transparent"
      shape="round"
      size={20}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000000",
            "borderRadius": 28,
            "display": "flex",
            "height": 28,
            "justifyContent": "center",
            "width": 28,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFBC11",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="rating-icon-4"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
